src/common/post.ts

```diff
@@ -329,3 +344,125 @@ export const insertCodeSnippetsFromUrl = async ({
     throw err;
   }
 };
+
+type ModeratedPostCdc = ChangeObject<SourcePostModeration>;
+
+export const updateModeratedPost = async (
+  con: ConnectionManager,
+  moderated: ModeratedPostCdc,
+): Promise<ModeratedPostCdc | null> => {
+  if (!moderated?.postId) {
+    logger.error('unable to update moderated post', { moderated });
+    return null;
+  }
+
+  const postParam = { id: moderated.postId };
+  const repo = con.getRepository(Post);
+
+  await repo.findOneOrFail({ where: { id: moderated.postId } });
+
+  const updatedPost: Partial<FreeformPost> = {
+    title: moderated.title,
+    titleHtml: moderated.titleHtml,
+  };
+
+  if (moderated.content) {
+    updatedPost.content = moderated.content;
+    updatedPost.contentHtml = moderated.contentHtml!;
+  }
+
+  if (moderated.image) {
+    updatedPost.image = moderated.image;
+  }
+
+  await repo.update(postParam, updatedPost);
+
+  return moderated;
+};
+
+export const getExistingPost = async (
+  manager: ConnectionManager,
+  cleanUrl: string,
+): Promise<Post | null> =>
+  await manager
+    .createQueryBuilder(Post, 'post')
+    .select(['post.id', 'post.deleted', 'post.visible'])
+    .where('post.url = :url OR post.canonicalUrl = :url', {
+      url: cleanUrl,
+    })
+    .getOne();
+
+export const processApprovedModeratedPost = async (
+  con: ConnectionManager,
+  moderated: ModeratedPostCdc,
+): Promise<ModeratedPostCdc | null> => {
+  if (!moderated) {
+    throw new Error('Moderated post is missing');
+  }
+
+  if (moderated.postId) {
+    return updateModeratedPost(con, moderated);
+  }
+
+  const {
+    title,
+    content,
+    contentHtml,
+    image,
+    createdById,
+    sourceId,
+    titleHtml,
+    externalLink,
+    sharedPostId,
+  } = moderated;
+
+  if (moderated.type === PostType.Freeform) {
+    const id = await generateShortId();
```

Comment:
Is this was we do now?
I think @ilasw mentioned it in another thread as well.

Reply:
This is for the actual post in the `post` table, for the moderation table, we generate it in DB level. If that is what you mean as I don't have reference to the thread.

Reply:
Ah maybe yes, this one we did now with shortId?

Reply:
I just saw from the other implementation where we create post we use this id. Wdyt, should we use uuid?

Reply:
I personally don't mind more up to @capJavert I think